mynav.cc

```diff
@@ -114,59 +98,42 @@ mynav_rd_line(const QString& line)
     }
   }
 
-  track_add_wpt(mynav_track, wpt);
+  track_add_wpt(track, wpt);
 }
 
+/***************************************************************************
+ *              entry points called by gpsbabel main process               *
+ ***************************************************************************/
 
-//***************************************************************************
-//           global callbacks called by gpsbabel main process
-//***************************************************************************
-
-static void
-mynav_rd_init(const QString& fname)
+void
+MyNavFormat::rd_init(const QString& fname)
 {
-  fin = gbfopen(fname, "rb", MYNAME);
-  mynav_track = route_head_alloc();
-  track_add_head(mynav_track);
+  read_fname = fname;
 }
 
-static void
-mynav_rd_deinit()
+void
+MyNavFormat::rd_deinit()
 {
-  gbfclose(fin);
+  read_fname.clear();
 }
 
-static void
-mynav_rd()
+void
+MyNavFormat::read()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/manual/gpsbabel/gpsbabel/pullRequest?prid=4885950)